app/views/transferAgreement.scala.html

```diff
@@ -28,69 +39,73 @@ <h1 class="govuk-heading-l">Transfer agreement</h1>
             @inputSingleCheckbox(
-            transferAgreementForm("droAppraisalSelection"),
-            Symbol("_label") -> "I confirm that the Departmental Records Officer (DRO) has signed off on the appraisal and selection decision.",
-            Symbol("_value") -> "true",
-            Symbol("_smallCheckbox") -> true,
-            Symbol("_checkedOption") -> transferAgreementForm.data.getOrElse("droAppraisalSelection", ""),
-            Symbol("_requiredOption") -> true
+                transferAgreementForm("droAppraisalSelection"),
+                Symbol("_label") -> "I confirm that the Departmental Records Officer (DRO) has signed off on the appraisal and selection decision.",
+                Symbol("_value") -> "true",
+                Symbol("_smallCheckbox") -> true,
+                Symbol("_checkedOption") -> shouldOptionBeSelected("droAppraisalSelection"),
+                Symbol("_requiredOption") -> true,
+                Symbol("_disabledOption") -> isAlreadyCompleted.toString
             )
             @inputSingleCheckbox(
-            transferAgreementForm("droSensitivity"),
-            Symbol("_label") -> "I confirm that the Departmental Records Officer (DRO) has signed off on the sensitivity review.",
-            Symbol("_value") -> "true",
-            Symbol("_smallCheckbox") -> true,
-            Symbol("_checkedOption") -> transferAgreementForm.data.getOrElse("droSensitivity", ""),
-            Symbol("_requiredOption") -> true
+                transferAgreementForm("droSensitivity"),
+                Symbol("_label") -> "I confirm that the Departmental Records Officer (DRO) has signed off on the sensitivity review.",
+                Symbol("_value") -> "true",
+                Symbol("_smallCheckbox") -> true,
+                Symbol("_checkedOption") -> shouldOptionBeSelected("droSensitivity"),
+                Symbol("_requiredOption") -> true,
+                Symbol("_disabledOption") -> isAlreadyCompleted.toString
             )
             @inputSingleCheckbox(
                 transferAgreementForm("openRecords"),
                 Symbol("_label") -> "I confirm that all records are open and no Freedom of Information (FOI) exemptions apply to these records.",
                 Symbol("_value") -> "true",
                 Symbol("_smallCheckbox") -> true,
-                Symbol("_checkedOption") -> transferAgreementForm.data.getOrElse("openRecords", ""),
-                Symbol("_requiredOption") -> true
+                Symbol("_checkedOption") -> shouldOptionBeSelected("openRecords"),
+                Symbol("_requiredOption") -> true,
+                Symbol("_disabledOption") -> isAlreadyCompleted.toString
             )
         </div>
-        <!-- Buttons -->
         <div class="govuk-button-group">
-            <!-- Continue -->
             <button data-prevent-double-click="true" class="govuk-button" type="submit" data-module="govuk-button" role="button">
                 Continue
```

Comment:
Should this be a link rather than a form submission button when we're showing the "already completed" version?

Reply:
The issue is, with these new set of commits, the TA page now one i.e. I'm no longer using a separate `TAAlreadyCompleted` page so do you suppose I should maybe add some logic to display the button (if the form has not been submitted) or the link (if it has)?

Reply:
The other way to commonise the page is to have two separate templates, and then bring in the common components as partials. That's sometimes easier when there are a lot of differences between the two types of page, rather than lots of `if` statements. Would it be worth trying it here?

Give me a shout if that doesn't make sense.